### PR TITLE
Expand documentation

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR expands the documentation of the package:
- it adds a section about `GaudiConfig` in the "Main Classes" section,
- it creates a section called "How-To Guides" where:
  - the section about distributed training has been moved there,
  - it adds a section about single-hpu training and about good practices for making training faster

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ limitations under the License.
 
 🤗 Optimum Habana is the interface between the 🤗 Transformers library and [Habana's Gaudi processor (HPU)](https://docs.habana.ai/en/latest/index.html).
 It provides a set of tools enabling easy model loading and fine-tuning on single- and multi-HPU settings for different downstream tasks.
-The current release focuses on question answering and text classification and enables users to try other models for other tasks with only a few changes.
+The list of officially supported models and tasks is available [here](https://github.com/huggingface/optimum-habana#validated-models). Users can try other models and tasks with only few changes.
```
Comment (libinta) [inline README.md:24]: instead of supported, should we say sth different like "verified"?

File: docs/source/accelerate_training.mdx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+
+## Mixed-Precision Training
+
+Mixed-precision training enables to compute some operations using lighter data types to accelerate training.
+Habana Mixed Preicision (HMP) proposes to mix *fp32* and *bf16* operations.
+In order to do so, you must set `"use_habana_mixed_precision"` to `true` and `"hmp_opt_level"` to `"O1"` in the Gaudi configuration file.
+Then, you can specify which operators to compute in *bf16* with `"hmp_bf16_ops"` and which operators to compute in *fp32* with `"hmp_fp32_ops"`.
+
```
Comment (libinta) [inline docs/source/accelerate_training.mdx:48]: should we also mention about the default https://docs.habana.ai/en/latest/PyTorch/PyTorch_User_Guide/PT_Mixed_Precision.html#opt-level-o1

File: docs/source/index.mdx [hunk 3]
```diff
@@ -15,10 +15,10 @@ limitations under the License.
 -->
 
 
-# Optimum Habana
+# 🤗 Optimum Habana
 
 🤗 Optimum Habana is the interface between the 🤗 Transformers library and [Habana's Gaudi processor (HPU)](https://docs.habana.ai/en/latest/index.html). It provides a set of tools enabling easy model loading and fine-tuning on single- and multi-HPU settings for different downstream tasks.
-The current release focuses on question answering and text classification and enables users to try other models for other tasks with only a few changes.
+The list of officially supported models and tasks is available [here](validated-models). Users can try other models and tasks with only few changes.
```
Comment (libinta) [inline docs/source/index.mdx:21]: same here, should we change supported to sth else?

File: README.md (hunk 1 above)
Comment (regisss) [reply README.md:24]: I remember in the README we put "validated", shall I put this?

File: docs/source/index.mdx (hunk 3 above)
Comment (regisss) [reply docs/source/index.mdx:21]: Same reply with "validated", wdyt?

File: docs/source/accelerate_training.mdx (hunk 2 above)
Comment (regisss) [reply docs/source/accelerate_training.mdx:48]: Just to be sure I understood, you would like to have something like "O1 is the default value" right?

File: README.md (hunk 1 above)
Comment (libinta) [reply README.md:24]: yes

File: docs/source/accelerate_training.mdx (hunk 2 above)
Comment (libinta) [reply docs/source/accelerate_training.mdx:48]: what I mean is for option 01, if you dont specify, there are default list.
also, there is option 2
see https://docs.habana.ai/en/latest/PyTorch/PyTorch_User_Guide/PT_Mixed_Precision.html#opt-level-o2 for details

File: docs/source/index.mdx (hunk 3 above)
Comment (libinta) [reply docs/source/index.mdx:21]: yes

File: docs/source/accelerate_training.mdx (hunk 2 above)
Comment (regisss) [reply docs/source/accelerate_training.mdx:48]: Okay, I will mention the default lists.

For O2, I already talk about it in the doc of the Gaudi configuration (see `gaudi_config.mdx` in this PR). Here, since those are pieces of advice for making training as fast as possible, I prefer mentioning only O1.

Comment (libinta) [reply docs/source/accelerate_training.mdx]: ok

File: docs/source/accelerate_training.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+
+## Mixed-Precision Training
+
+Mixed-precision training enables to compute some operations using lighter data types to accelerate training.
+Habana Mixed Preicision (HMP) proposes to mix *fp32* and *bf16* operations.
+In order to do so, you must set `"use_habana_mixed_precision"` to `true` and `"hmp_opt_level"` to `"O1"` in the Gaudi configuration file.
+Then, you can specify which operators to compute in *bf16* with `"hmp_bf16_ops"` and which operators to compute in *fp32* with `"hmp_fp32_ops"`.
+If these operators are not specified, their default values are set to be the ones written in the [Gaudi configuration file of BERT](https://huggingface.co/Habana/bert-large-uncased-whole-word-masking/blob/main/gaudi_config.json), which is a good starting point for applying HMP:
```
Comment (libinta) [inline docs/source/accelerate_training.mdx:48]: did you have a default list for each model? our default list is Default BF16 list = [addmm, bmm, conv1d, conv2d, conv3d, dot, mm, mv]

Default FP32 list = [batch_norm, cross_entropy, log_softmax, softmax, nll_loss, topk]
see https://docs.habana.ai/en/latest/PyTorch/PyTorch_User_Guide/PT_Mixed_Precision.html#opt-level-o1

Comment (regisss) [reply docs/source/accelerate_training.mdx:48]: Yes, the default lists are defined in `gaudi_configuration.py`, you can see them [here](https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/gaudi_configuration.py). Those are basically the list of ops we use for BERT/DistilBERT/RoBERTa/ALBERT.

You want me to modify them so that they are the same as in Synapse?

Comment (libinta) [reply docs/source/accelerate_training.mdx:48]: no, what you have is fine since it's for transformer related.

**Review by libinta — APPROVED** [review]

File: docs/source/accelerate_training.mdx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "log_softmax"
+]
+```
+
+
+## Custom Operators
+
+### Fused ADAM
+
+Habana provides a [custom fused ADAM implementation](https://docs.habana.ai/en/v1.5.0/PyTorch/Model_Optimization_PyTorch/Custom_Ops_PyTorch.html#custom-optimizers).
```
Comment (philschmid) [inline docs/source/accelerate_training.mdx:78]: Is there a way to point to the latest docs an not `1.5.0` ?

File: docs/source/accelerate_training.mdx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+"hmp_fp32_ops": [
+    "embedding",
+    "nll_loss",
+    "log_softmax"
+]
+```
+
+
+## Custom Operators
+
```
Comment (philschmid) [inline docs/source/accelerate_training.mdx:75]: Would a Sentence in between to have not two headlines following each other. Maybe 1 liner what custom operators are. 

We could also link here to https://docs.habana.ai/en/latest/PyTorch/PyTorch_CustomOp_API/page_index.html. 

QQ: are the regular PyTorch Ops custom Ops for habana or would you define them as regular Ops?

File: docs/source/accelerate_training.mdx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+## Mixed-Precision Training
+
+Mixed-precision training enables to compute some operations using lighter data types to accelerate training.
+Habana Mixed Preicision (HMP) proposes to mix *fp32* and *bf16* operations.
+In order to do so, you must set `"use_habana_mixed_precision"` to `true` and `"hmp_opt_level"` to `"O1"` in the Gaudi configuration file.
+Then, you can specify which operators to compute in *bf16* with `"hmp_bf16_ops"` and which operators to compute in *fp32* with `"hmp_fp32_ops"`.
+If these operators are not specified, their default values are set to be the ones written in the [Gaudi configuration file of BERT](https://huggingface.co/Habana/bert-large-uncased-whole-word-masking/blob/main/gaudi_config.json), which is a good starting point for applying HMP:
+```
+"hmp_bf16_ops": [
```
Comment (philschmid) [inline docs/source/accelerate_training.mdx:50]: We should link somewhere to https://docs.habana.ai/en/latest/PyTorch/Pytorch_Operators/Pytorch_Operators.html

Maybe add a section before "Mixed-Precision Training" with "Supported PyTorch operators"?

**Review by philschmid — APPROVED** [review]
Very nice!🚀 I added some comments.

File: docs/source/accelerate_training.mdx (hunk 5 above)
Comment (regisss) [reply docs/source/accelerate_training.mdx]: So in the end I changed all the links so that they point to the latest version as you suggested.

File: docs/source/accelerate_training.mdx (hunk 6 above)
Comment (regisss) [reply docs/source/accelerate_training.mdx:80]: I just added a small introductory paragraph for custom ops with a reference to this link.

I consider them as regular ops. To me the only custom ops are the ones listed here: https://docs.habana.ai/en/latest/PyTorch/Model_Optimization_PyTorch/Custom_Ops_PyTorch.html

File: docs/source/accelerate_training.mdx (hunk 7 above)
Comment (regisss) [reply docs/source/accelerate_training.mdx:55]: I added a warning in the "Mixed-Precision Training" section to mention this.

File: docs/source/accelerate_training.mdx (hunk 5 above)
Comment (philschmid) [reply docs/source/accelerate_training.mdx]: Totally agree that aligned version would be nice but that's something you easily for get to update. And i think the goal is to always work with the latest synapseAI version

Comment (regisss) [reply docs/source/accelerate_training.mdx]: Yes I agree with you. There should just be a few days between the release of a new version of SynapseAI and its added support in the library, so overall linking to the latest version of the doc is the most convenient way to deal with this.
